lib/ecl/ecl_sum_vector.c

```diff
@@ -58,6 +58,27 @@ ecl_sum_vector_type * ecl_sum_vector_alloc(const ecl_sum_type * ecl_sum){
     return ecl_sum_vector;
 }
 
+static void ecl_sum_vector_add_invalid_key(ecl_sum_vector_type * vector, const char * key) {
+  int_vector_append(vector->node_index_list, -1);
+  bool_vector_append(vector->is_rate_list, false);
+  stringlist_append_copy(vector->key_list, key);
+}
+
+
+
+ecl_sum_vector_type * ecl_sum_vector_alloc_layout_copy(const ecl_sum_vector_type * vector, const ecl_sum_type * ecl_sum) {
+  ecl_sum_vector_type * new_vector = ecl_sum_vector_alloc(ecl_sum);
```

Comment:
Shouldn't this use `vector->ecl_sum` and not take `ecl_sum` as an argument?  Or perhaps I don't understand this function.   Can you add a comment?

Reply:
> Can you add a comment?

Yes - I will add a comment.